packages/react/src/components/ButtonSet/ButtonSet.tsx

```diff
@@ -1,36 +1,127 @@
 /**
- * Copyright IBM Corp. 2016, 2023
+ * Copyright IBM Corp. 2016, 2024
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 import { ForwardRefReturn } from '../../types/common';
+import useIsomorphicEffect from '../../internal/useIsomorphicEffect';
 
 export interface ButtonSetProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * fluid: button set resize to the size of the container up to a maximum dependant on the
+   * number of buttons. Overrides `stacked` property.
+   */
+  fluid?: boolean;
   /**
    * Specify the button arrangement of the set (vertically stacked or
    * horizontal)
    */
   stacked?: boolean;
 }
 
+const fluidMultiples = ['none', 'single', 'double', 'triple', 'quadruple'];
+const fluidMultiple = (buttonArray) => {
+  if (buttonArray.length >= fluidMultiples.length) {
+    // too many buttons
+    return 'many';
+  } else {
+    return fluidMultiples[buttonArray.length];
```

Comment:
Is there an opportunity to avoid this altogether by instead determining the children count [in the styles directly with `:has()`](https://www.matuzo.at/blog/2022/counting-children)?